File: src/load.js
import mcf from "./mcf.js"

const folder = mcf.load;
const { scoreboardObjectives, scoreboardPlayers, addTeams, setupData, returnPlayer, anonymous: a } = folder;

const scoreboards = {
	ip_score: "Infinite Parkour - Distance (M)",
	ip_highscore: "Infinite Parkour - High Score",
	math: "General math calculations",
	const: "Constant values",
	ipe_index: "Index",
	ip_data: "Data",
	ip_last_online: "Last Online",
	ip_lane: "Lane",
	ip_flight_cooldown: "Flight Cooldown",
	ip_anim: "Animation Frame Counter for Entities",
};

// Sets up things to be used generally
scoreboardObjectives`
	${Object.entries(scoreboards).map(([name, display]) => `scoreboard objectives add ${name} dummy "${display}"`).join("\n")}
`

const consts = [
	-1, 1, 2, 4, 5,
	31, 64,
	100, 128, 180, 200,
	1024, 8000,
	10000, 16000,
];
scoreboardPlayers`
	${consts.map(n => `scoreboard players set ${n} const ${n}`).join("\n")}
`


gamerules`
#Below is only for alpha release
gamerule doMobLoot false
gamerule fallDamage false
`



const teams = {
	Highscore: 'gold',
	ParkourPlayers: 'yellow',
	infpar_yellow: 'yellow',
	infpar_green: 'green',
	infpar_dark_green: 'dark_green',
	infpar_editor: 'white',
};

addTeams`
	${Object.entries(teams).map(([team, color]) => `
		team add ${team}
		team modify ${team} color ${color}
		team modify ${team} collusionRule never
	`).join("\n")}
`

setupData`
	execute unless data storage infinite_parkour:player_data players run data modify storage infinite_parkour:player_data players set value []
	function infinite_parkour:load_base_packs
`

returnPlayer`
	execute as @a at @s if dimension infinite_parkour:lane in minecraft:overworld run ${a`
		tp @s 0 0 0
		function infinite_parkour:freeplay
	`}
	execute as @a at @s if dimension infinite_parkour:editor in minecraft:overworld run ${a`
		tp @s 0 0 0
		function infinite_parkour:editor
	`}
	execute in infinite_parkour:lane run ${a`
		forceload add 0 0 0 0
		setblock 0 0 0 minecraft:barrel
	`}
`;

//say §aFinished loading!

File: src/load.py
from python import *

# === Helper functions ===

def add_objectives(objs):
    for name, title in objs:
        c(f'scoreboard objectives add {name} dummy {{text:"{title}"}}')

def set_constants(vals):
    for val in vals:
        c(f"scoreboard players set {val} const {val}")

def add_teams(team_defs):
    for name, text, color in team_defs:
        display = f' {{text:"{text}"}}' if text else ''
        c(f"team add {name}{display}")
        c(f"team modify {name} color {color}")

def set_collision_rules(team_names):
    for name in team_names:
        c(f"team modify {name} collisionRule never")

def batch(commands):
    for cmd in commands:
        c(cmd)

# === Execution ===

add_objectives([
    ("ip_score", "Infinite Parkour - Distance (M)"),
    ("ip_highscore", "Infinite Parkour - High Score"),
    ("const", "Constant values"),
    ("ipe_index", "Index"),
    ("ip_data", "Data"),
    ("ip_last_online", "Last Online"),
    ("ip_lane", "Lane"),
    ("ip_flight_cooldown", "Flight Cooldown"),
    ("ip_anim", "Animation Frame Counter for Entities"),
])

set_constants([-1, 1, 2, 4, 5, 31, 64, 100, 128, 180, 200, 1024, 8000, 10000, 16000])

batch([
    "gamerule commandModificationBlockLimit 999999999",
    "gamerule doMobLoot false",
    "gamerule fallDamage false",
])

add_teams([
    ("Highscore",        "InLobby",     "gold"),
    ("ParkourPlayers",   "InParkour",   "yellow"),
    ("infpar_yellow",    None,          "yellow"),
    ("infpar_green",     None,          "green"),
    ("infpar_dark_green",None,          "dark_green"),
    ("infpar_editor",    "InEditor",    "white"),
])

set_collision_rules(["Highscore", "ParkourPlayers", "infpar_editor"])

c("execute unless data storage infinite_parkour:player_data players run data modify storage infinite_parkour:player_data players set value []")

player_dim_logic = [
    ("infinite_parkour:lane", "tp @s 0 0 0", "function infinite_parkour:freeplay"),
    ("infinite_parkour:editor", "tp @s 0 0 0", "function infinite_parkour:editor"),
]
for dim, tp_cmd, func_cmd in player_dim_logic:
    base = f"execute as @a at @s if dimension {dim} in minecraft:overworld run"
    c(f"{base} {tp_cmd}")
    c(f"{base} {func_cmd}")

batch([
    "execute in infinite_parkour:lane run forceload add 0 0 0 0",
    "execute in infinite_parkour:lane run setblock 0 0 0 minecraft:barrel",
])
import src.load_base_packs
src.load_base_packs.main()
#c("say §aFinished loading!")